stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -349,4 +349,31 @@ public void onCancel() {
     public void onComplete() {
     }
   }
+
+  public static void unimplementedStreamResponse(StreamObserver responseObserver) {
+    responseObserver.onError(Status.UNIMPLEMENTED.asException());
+  }
+
+  public static class UnimplementedStreamObserver<ReqT> implements StreamObserver<ReqT> {
+    private final StreamObserver responseObserver;
+
+    public UnimplementedStreamObserver(StreamObserver responseObserver) {
+      this.responseObserver = responseObserver;
+    }
+
+    @Override
+    public void onNext(ReqT value) {
+      unimplementedStreamResponse(responseObserver);
```

Comment:
Hmm... that's unexpected. Why aren't we able to use unimplementedStreamResponse() immediately and use a NOOP StreamObserver?